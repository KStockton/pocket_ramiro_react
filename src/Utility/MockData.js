export const mockPartsResouce = [
  {
      "id": 1,
      "name": "lever 2000",
      "inventory": 1,
      "active": true,
      "created_at": "2019-07-16T13:15:31.885Z",
      "updated_at": "2019-07-16T13:16:16.146Z"
  },
  {
      "id": 2,
      "name": "Tank patch",
      "inventory": 1,
      "active": true,
      "created_at": "2019-07-16T13:18:09.048Z",
      "updated_at": "2019-07-16T13:18:09.048Z"
  }
]