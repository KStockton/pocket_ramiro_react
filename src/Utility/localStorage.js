export const loadState = () => {
  return localStorage.getItem('reduxState') ? JSON.parse(localStorage.getItem('reduxState')) : {}
};

export const saveState = (state) => {
  const serializedState = JSON.stringify(state);
  localStorage.setItem('reduxState', serializedState);
};

export const clearState = () => {
  localStorage.removeItem('reduxState');
};