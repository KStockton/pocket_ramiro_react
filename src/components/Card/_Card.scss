.header-icon-container {
  @include displayFlex(null, null, center, null);
  border-bottom: 1px #00000026 solid;
  padding-bottom: 10px;
}

.header-icon {
  margin-right: 5px;
}

.low {
  color: yellow;
}
.medium {
  color: orange;
}
.high {
  color: red;
}
.urgent {
  color: purple;
}
.safety {
  color: teal;
}

.card {
  border: 2px solid #d6c4c49e;
  border-radius: 25px;
  margin: 22px 0px 25px -3px;
  min-height: 26rem;
  box-shadow: 0px 5px 7px #2d2b2dc4;
  padding: 10px 35px;
  position: relative;
  overflow: hidden;
  z-index: 10;
  max-width: 18rem;

  header {
    @include displayFlex(space-around, null, null, null)
  }

  p {
    font-family: 'Nunito', sans-serif;
  }

}

.card-notes {
  width: 10vw;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}


