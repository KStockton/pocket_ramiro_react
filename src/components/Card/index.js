import React from 'react';
import { Link } from 'react-router-dom';


export const Card = (props) => {
  const {notes, priority, created_at, resource_type_id, name, cost, id} = props.item;
  const calcDaysSinceCreation = (t) => Math.floor(t / (24 * 60 * 60 * 1000));
  const days = calcDaysSinceCreation(Math.abs(new Date() - new Date(created_at)));

  return (
    <section className='' >
      <header>
        {
          <p> #{id}</p>
        }

        { 
          priority && <p className="header-icon-container">
            <i className={`material-icons header-icon ${priority}`}>
            flag
            </i>Priority: {priority}
          </p> 
        } 

        { 
          resource_type_id && <p className="header-icon-container">
            <i className={`material-icons header-icon ${name}`}>
            business
            </i>Name: {name}
          </p> 
        }
        
        { 
          cost && <p className="header-icon-container">
            <i className={`material-icons header-icon ${name}`}>
            business
            </i>Cost: {cost}
          </p> 
        }

        <p className="header-icon-container">
          <i className={`material-icons header-icon`}>
            drafts
          </i>Opened: {created_at.substr(0, 10)}
        </p>

        <p className="header-icon-container">
          <i className={`material-icons header-icon`}>
            access_time
          </i>{`${days} days since ticket opened`}
        </p>

        { 
          resource_type_id &&
            <Link to={{
              pathname: '/resources/parts',
              state: {
                resourceId: resource_type_id
              }
            }}>
              <i className="material-icons">
                more_horiz
              </i>
            </Link>
        }

      </header>
      { notes && <p className='card-notes' >
        <i class="material-icons header-icon">
          notes
        </i>
        Notes: {notes} </p>}
    </section>
  );
};

export default Card;