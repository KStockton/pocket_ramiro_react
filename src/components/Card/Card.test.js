import React from 'react';
import { shallow } from 'enzyme'
import Card from './';
import * as MD from '../../Utility/MockData'

describe('Card', () => {
  let wrapper;

  beforeEach(() => {
 
    wrapper = shallow(<Card
    ticket={ MD.mockTicket}
    />)
  });

  it('should match the snapshot', () => {
    
    wrapper.setProps({created_at: MD.mockResources[0].created_at})
    expect(wrapper).toMatchSnapshot()
  });

});
 