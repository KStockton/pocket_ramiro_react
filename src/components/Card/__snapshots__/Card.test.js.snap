// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<section
  className=""
>
  <header>
    <p>
       #
      11
    </p>
    <p
      className="header-icon-container"
    >
      <i
        className="material-icons header-icon medium"
      >
        flag
      </i>
      Priority: 
      medium
    </p>
    <p
      className="header-icon-container"
    >
      <i
        className="material-icons header-icon"
      >
        drafts
      </i>
      Opened: 
      2019-06-16
    </p>
    <p
      className="header-icon-container"
    >
      <i
        className="material-icons header-icon"
      >
        access_time
      </i>
      34 days since ticket opened
    </p>
  </header>
  <p
    className="card-notes"
  >
    Notes: 
    Ut enim
  </p>
</section>
`;

exports[`Card should match the snapshot if a resource is passed as props 1`] = `
<section
  className=""
>
  <header>
    <p>
       #
      1
    </p>
    <p
      className="header-icon-container"
    >
      <i
        className="material-icons header-icon Bright Tank 1"
      >
        business
      </i>
      Name: 
      Bright Tank 1
    </p>
    <p
      className="header-icon-container"
    >
      <i
        className="material-icons header-icon Bright Tank 1"
      >
        business
      </i>
      Cost: 
      5000
    </p>
    <p
      className="header-icon-container"
    >
      <i
        className="material-icons header-icon"
      >
        drafts
      </i>
      Opened: 
      2019-07-16
    </p>
    <p
      className="header-icon-container"
    >
      <i
        className="material-icons header-icon"
      >
        access_time
      </i>
      3 days since ticket opened
    </p>
    <Link
      to={
        Object {
          "pathname": "/resources/parts",
          "state": Object {
            "resourceId": 1,
          },
        }
      }
    >
      <i
        className="material-icons"
      >
        more_horiz
      </i>
    </Link>
  </header>
</section>
`;
